src/interactive-auth.js

```diff
@@ -235,13 +244,39 @@ InteractiveAuth.prototype = {
             throw new Error("submitAuthDict() called before attemptAuth()");
         }
 
+        if (!background && this._busyChangedCallback) {
+            this._busyChangedCallback(true);
+        }
+
+        // if we're currently trying a request, wait for it to finish
+        // as otherwise we can get multiple 200 responses which can mean
+        // things like multiple logins for register requests.
+        // (but discard any expections as we only care when its done,
+        // not whether it worked or not)
+        while (this._submitPromise) {
+            try {
+                await this._submitPromise;
+            } catch (e) {
+            }
+        }
+
         // use the sessionid from the last request.
         const auth = {
             session: this._data.session,
         };
         utils.extend(auth, authData);
 
-        await this._doRequest(auth, background);
+        try {
+            // NB. the 'background' flag is deprecated by the busyChanged
```

Comment:
Ah, when I mentioned documenting the deprecation, I was actually thinking of the `background` param docs on this `submitAuthDict` method. Oh but maybe this `submitAuthDict` is only called internally by other method of this object...?

Reply:
Ah, so, it's a little weird in that the `submitAuthDict` is public and called from externally, but never with the `background` param - also, the background param is still used at this level, just not passed out to the app.